BinaryClock/Display.cpp

```diff
@@ -46,17 +46,53 @@ void Display::print_static_text(const String &text)
     delay(1000);
 }
 
+ // void Display::print_time_binary(const Time &val)
+ // {
+ //     MATRIX.clear();
+ //
+ //     String digits = val.to_binary_string();
+ //     for (int i = 0; i < digits.length(); i++) {
+ //         if (digits.charAt(i) == '1') {
+ //             unsigned height = MATRIX.height();
+ //             MATRIX.drawFastVLine(i, 0, height, LED_ON);
+ //         }
+ //     }
+ //
+ //     MATRIX.writeDisplay();
+ // }
+
 void Display::print_time_binary(const Time &val)
 {
-    MATRIX.clear();
+   MATRIX.clear();
 
-    String digits = val.to_binary_string();
-    for (int i = 0; i < digits.length(); i++) {
-        if (digits.charAt(i) == '1') {
-            unsigned height = MATRIX.height();
-            MATRIX.drawFastVLine(i, 0, height, LED_ON);
-        }
-    }
+   String digits = val.to_binary_string();
+   for (int i = 0; i < 5; i++)
+   {
+       if (digits.charAt(i) == '1'){
+       unsigned height = MATRIX.height();
+       MATRIX.drawFastVLine(i, 0, height, LED_ON);
```

Comment:
By "right aligned," I mean that *all* the columns, including the colon you added, should all be shifted to the right edge of the screen instead of the left.

I noticed that the for loop counter `i` still starts at zero and increments upward, and in line 73 it's used as the X coordinate for the `VLine` just like the original code. The coordinate system's X axis increases from left to right, so therefore this code is not right aligned. Perhaps in every `drawFastVLine(i, ...)` invocation you could replace it with `drawFastVLine(i + some_offset)`?